src/base/bittorrent/session.cpp

```diff
@@ -1416,6 +1423,25 @@ void Session::loadLTSettings(lt::settings_pack &settingsPack)
 
     settingsPack.set_bool(lt::settings_pack::allow_multiple_connections_per_ip, multiConnectionsPerIpEnabled());
 
+#if (LIBTORRENT_VERSION_NUM >= 10206)
+    settingsPack.set_bool(lt::settings_pack::piece_extent_affinity, usePieceExtentAffinity());
+
+    switch (sslVersion()) {
+    case SSLVersion::TLS11:
+        settingsPack.set_int(lt::settings_pack::ssl_version, lt::settings_pack::tls11);
+        break;
+    case SSLVersion::TLS12:
+    default:
+        settingsPack.set_int(lt::settings_pack::ssl_version, lt::settings_pack::tls12);
+        break;
+#if (BOOST_VERSION >= 106900)
```

Comment:
@arvidn
IMO libtorrent should have defined a macro for its clients to use based on whether libtorrent was compiled with boost >=1.69. It doesn't make sense to let clients to look for BOOST_VERSION here.

Reply:
I'm open to concrete suggestions. In general I prefer solutions that don't require more configuration macros. I was considering leaving the `tls13` enum in, and just have it silently mean `tls12` on an older boost. But that seemed risky.

However, what about having an enum value `latest`, that refers to the most recent version supported?

Reply:
I like the idea of latest, but how will Joe qBit know whether it implies TLS 1.2 or TLS 1.3? 

I know the current version is a bit ugly, but the idea was to not display TLS 1.3 if it wasn't supported so people aren't under the assumption that they are using TLS 1.3.

Reply:
Just for the record, I was thinking of something like this:
https://en.cppreference.com/w/cpp/feature_test#Library_features
But I'm not clear of its downside nor don't know if it is too much for a library to maintain one.

>I was considering leaving the tls13 enum in, and just have it silently mean tls12 on an older boost. But that seemed risky.

yeah that doesn't sound ideal.

>However, what about having an enum value latest, that refers to the most recent version supported?

You mean `enum ssl_version_t` contains an additional value `latest`?
And that should help, I imagine we can compare `ssl_version_t::latest` and `ssl_version_t::tls12` and if they are the same we can hide "TLS 1.3" option and show it otherwise.

Reply:
well, it would still involve checking `BOOST_VERSION`, since your code would still have to mention `tls13`

Reply:
I believe @Chocobo1 is trying to completely avoid checking BOOST_VERSION, hence the request for another configuration macro.

Wouldn't the macro be as simple as this?:
#if (BOOST_VERSION >= 106900)
#define TORRENT_SUPPORTS_TLS_13
#endif

If so, I can only see that being cleaner for programs implementing libtorrent, unless there's maintainability issues or other drawbacks I'm unaware of.

Reply:
I added `TORRENT_USE_TLS13` to be defined as either 0 or 1

Reply:
Thank you! I'll update this PR later today.